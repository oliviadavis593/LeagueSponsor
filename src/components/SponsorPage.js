import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import config from '../config';
import Logo from '../imgs/leaguelogo.png';
import '../styles/SponsorPage.css';

class SponsorPage extends Component {
    constructor() {
        super();
        this.state = {
            parameters: {},
            leagues: [],
            error: null,
        }
    }

    hasRequiredParams = parameters => {
        const { latitude, longitude, radius, budget } = this.state.parameters;
        if (latitude && longitude && radius && budget) {
            return true; 
        }
        return false; 
    }

    handleSumbit = event => {
        event.preventDefault();
        const parameters = {};

        Object.values(event.target.elements).forEach(input => {
            if (input.tagNaem === 'INPUT') {
                parameters[input.name] = input.value; 
            }
        });

        this.setState({...this.state, parameters });
    }

    componentDidUpdate(_, prevState) {
        if (this.hasRequiredParams()) {
            if (
                prevState.parameters.latitude !== this.state.parameters.latitude ||
                prevState.parameters.longitude !== this.state.parameters.longitude ||
                prevState.parameters.radius !== this.state.parameters.radius ||
                prevState.parameters.budget !== this.state.parameters.budget
            ) {
                const { latitude, longitude, radius, budget } = this.state.parameters; 
                fetch(`${config.API_ENDPOINT}/api/leagues?latitude=${latitude}&longitude=${longitude}&radius=${radius}&budget=${budget}`)
                    .then(res => {
                        if (res.ok) {
                            return res.json();
                        }
                    })
                    .then(leagues => {
                        this.setState({ leagues, error: null })
                    })
                    .catch(error => {
                        this.setState({
                            ...this.state, 
                            error: 'Failed to fetch the leagues'
                        });
                    });
            }
        } 
    }

    render() {
        return(
            <div>
                <NavLink to='/dashboard'>
                    <img src={Logo} alt='ls-logo' className='ls-logo' />
                </NavLink>
                <main>
                    <form
                    className='ls-sponsor__form'
                    onSubmit={this.handleSumbit}
                    >
                        <fieldset>
                            <legend>
                                <h1>Sponsor Search</h1>
                            </legend>
                            <label htmlFor='budget'>Total Budget:</label>
                            <input 
                            id='budget'
                            type='number'
                            name='budget'
                            />

                            <label htmlFor='latitude'>Latitude:</label>
                            <input 
                            id='latitude'
                            type='number'
                            name='latitude'
                            />

                            <label htmlFor='longitude'>Longitude:</label>
                            <input 
                            id='longitude'
                            type='number'
                            name='longitude'
                            />

                            <label htmlFor='radius'>Radius:</label>
                            <select name='radius' id='radius'>
                                <option value='5'>5mi</option>
                            </select>
                        </fieldset>
                        <div className='sponsor__button'>
                            <button
                            className='btn'
                            >
                                Find
                            </button>
                        </div>
                        {this.state.error && <p>{this.state.error}</p>}
                    </form>
                </main>
            </div>
        )
    }
}

export default SponsorPage; 