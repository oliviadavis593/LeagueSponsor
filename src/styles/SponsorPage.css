.ls-sponsor__form {
    width: 300px;
    height: 500px; 
    margin: 0 auto; 
    border: 1px solid #233142;
    background-color: #233142; 
    color: #fff;
}

#radius {
    margin-left: 10px; 
}

.btn {
    width: 80px; 
    height: 35px; 
    margin-top: 30px; 
    font-family: 'Oswald', sans-serif;
    background-color: #def3fa;
    border: none; 
    font-size: 20px;
}

.btn:hover {
    color: #fbc531;
    transition: 0.7s; 
}