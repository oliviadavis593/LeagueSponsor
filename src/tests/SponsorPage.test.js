import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SponsorPage from '../components/SponsorPage';

describe('SponsorPage Component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><SponsorPage /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><SponsorPage /></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})