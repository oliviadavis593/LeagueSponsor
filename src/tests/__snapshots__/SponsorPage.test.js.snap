// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorPage Component renders the UI as expected 1`] = `
<div>
  <a
    aria-current={null}
    href="/dashboard"
    onClick={[Function]}
  >
    <img
      alt="ls-logo"
      className="ls-logo"
      src="leaguelogo.png"
    />
  </a>
  <main>
    <form
      className="ls-sponsor__form"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          <h1>
            Sponsor Search
          </h1>
        </legend>
        <label
          htmlFor="budget"
        >
          Total Budget:
        </label>
        <input
          id="budget"
          name="budget"
          placeholder="1750"
          type="number"
        />
        <label
          htmlFor="latitude"
        >
          Latitude:
        </label>
        <input
          id="latitude"
          name="latitude"
          placeholder="17"
          type="number"
        />
        <label
          htmlFor="longitude"
        >
          Longitude:
        </label>
        <input
          id="longitude"
          name="longitude"
          placeholder="20"
          type="number"
        />
        <label
          htmlFor="radius"
        >
          Radius:
        </label>
        <select
          id="radius"
          name="radius"
        >
          <option
            value="5"
          >
            5mi
          </option>
        </select>
      </fieldset>
      <div
        className="sponsor__button"
      >
        <button
          className="btn"
        >
          Find
        </button>
      </div>
    </form>
  </main>
</div>
`;
